server/peers.py

```diff
@@ -77,8 +77,8 @@ def connection_made(self, transport):
         proto_ver = (version.PROTOCOL_MIN, version.PROTOCOL_MAX)
         self.send_request(self.on_version, 'server.version',
                           [version.VERSION, proto_ver])
-        self.send_request(self.on_peers_subscribe, 'server.peers.subscribe')
         self.send_request(self.on_features, 'server.features')
+        self.send_request(self.on_peers_subscribe, 'server.peers.subscribe')
 
```

Comment:
What I was concerned about was the order that the requests are popped and processed. I moved the on_peers_subscribe to be last because it needs to know if the on_features result was failure. Before I moved it, on_peers_subscribe would fire before "incorrect genesis hash" was detected.